tufrepo/cli.py

```diff
@@ -248,18 +248,23 @@ def add_delegation(
     paths: Tuple[str],
     hash_prefixes: Tuple[str],
 ):
-    """Delegate from ROLE to DELEGATE"""
+    """Delegate from ROLE to DELEGATE.
+
+    Note: if ROLE contains succinct hash bin delegations it will be overwritten.
+    """
 
     _paths = list(paths) if paths else None
     _prefixes = list(hash_prefixes) if hash_prefixes else None
 
-    targets: Targets
-    with ctx.obj.repo.edit(ctx.obj.role) as targets:
-        if targets.delegations is None:
-            targets.delegations = Delegations({}, {})
+    tar: Targets
+    with ctx.obj.repo.edit(ctx.obj.role) as tar:
+        if tar.delegations is None or tar.delegations.succinct_roles is not None:
+            tar.delegations = Delegations({}, {})
 
         role = DelegatedRole(delegate, [], 1, terminating, _paths, _prefixes)
-        targets.delegations.roles[role.name] = role
+        assert tar.delegations.roles is not None
```

Comment:
It was suggested by an extension I have.
It says: `Object of type \"None\" is not subscriptable`.

Usually, it gives good suggestions, but I am not sure about this one.
Here as we know that if a `targets` didn't had a delegation one will be created or if did have with succinct hash delegation it will be overwritten. 
We also know that one of `roles` or `succinct_roles` should be set in  a `Delegation` object meaning this is a little extra check.
Will remove it.